### PR TITLE
Don't stop headless server on SIGINT on windows (go-delve#1745)

### PR DESCRIPTION
**Pull request by nd**
This allows to execute and debug signal handlers in inferior.

### CODE REVIEW

File: cmd/dlv/cmds/commands.go [hunk 1]
```diff
@@ -675,9 +675,25 @@ func execute(attachPid int, processArgs []string, conf *config.Config, coreFile
...
+			// as SIGINT to delve. Ignore it instead of stopping the server
+			// in order to be able to debug signal handlers.
+			go func() {
+				for {
+					select {
+					case <-ch:
+					}
+				}
+			}()
+			select {
```
Comment (ProximaB) [inline cmd/dlv/cmds/commands.go:689]: @aarzilli  Could it be just one statement  `<-disconnectChan` instead of wrapping it in select?

Comment (aarzilli) [reply cmd/dlv/cmds/commands.go:689]: Sounds right.

Comment (ProximaB) [reply cmd/dlv/cmds/commands.go:689]: I'll make the pull request, alright?

Comment (aarzilli) [reply cmd/dlv/cmds/commands.go:689]: Sure.

**Pull request merged**